src/Tribe/Commerce/Currency.php

```diff
@@ -580,4 +580,256 @@ public function get_symbols_for_codes( $codes ) {
+			'thousands_sep'      => $this->get_currency_thousands_sep( $provider ),
+			'number_of_decimals' => $this->get_number_of_decimals( $provider ),
+		];
+	}
+
+	/**
+	 * Get Formatted Currency According to a Provider
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $amount   the amount to format
+	 * @param int         $post_id  The id of the post with tickets
+	 * @param string|null $provider The ticket provider class name
+	 *
+	 * @return mixed|void
+	 */
+	public function get_formatted_currency( $amount, $post_id, $provider = null ) {
+
+		$currency = $this->get_currency_by_provider( $post_id, $provider );
+
+		// Format the amount
+		if ( $currency['decimal_point'] == ',' && false !== ( $sep_found = strpos( $amount, $currency['decimal_point'] ) ) ) {
+			$whole  = substr( $amount, 0, $sep_found );
+			$part   = substr( $amount, $sep_found + 1, ( strlen( $amount ) - 1 ) );
+			$amount = $whole . '.' . $part;
+		}
+
+		// Strip , from the amount (if set as the thousands separator)
+		if ( $currency['thousands_sep'] == ',' && false !== ( $found = strpos( $amount, $currency['thousands_sep'] ) ) ) {
+			$amount = str_replace( ',', '', $amount );
+		}
+
+		// Strip ' ' from the amount (if set as the thousands separator)
+		if ( $currency['thousands_sep'] == ' ' && false !== ( $found = strpos( $amount, $currency['thousands_sep'] ) ) ) {
+			$amount = str_replace( ' ', '', $amount );
+		}
+
+		if ( empty( $amount ) ) {
+			$amount = 0;
+		}
+
+		$formatted = number_format( $amount, $currency['number_of_decimals'], $currency['decimal_point'], $currency['thousands_sep'] );
+
+		/**
+		 * Filter the Formatted Currency
+		 *
+		 * @since TBD
+		 *
+		 * @param string $formatted the formatted amount
+		 * @param int    $amount    the original amount to be formatted
+		 * @param array  $currency  an array of currency formatting details
+		 */
+		return apply_filters( 'tribe_format_amount', $formatted, $amount, $currency );
+	}
+
+	/**
+	 * Get Formatted Currency According to a Provider with Symbol
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $amount   the amount to format
+	 * @param int         $post_id  The id of the post with tickets
+	 * @param string|null $provider The ticket provider class name
+	 * @param boolean     $html     whether to return with html wrap
+	 *
+	 * @return mixed
+	 */
+	public function get_formatted_currency_with_symbol( $amount, $post_id, $provider = null, $html = true ) {
+
+		$amount          = $this->get_formatted_currency( $amount, $post_id, $provider );
+		$amount          = $html ? '<span class="tribe-amount">' . $amount . '</span>' : $amount;
+		$currency        = $this->get_currency_by_provider( $post_id, $provider );
+		$currency_symbol = $html ? '<span class="tribe-currency-symbol">' . $currency['symbol'] . '</span>' : $currency['symbol'];
+
+		$formatted = $html ? '<span class="tribe-formatted-currency-wrap tribe-currency-prefix">' . $currency_symbol . $amount . '</span>' : $currency_symbol . $amount;
```

Comment:
Feels like this would be easier to follow as nested ifs?
Also means we only create the string once - right now for `postfix` we create it and then overwrite it?